spec/codebreaker/game_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module Codebreaker
+  RSpec.describe Game do
+    subject(:game) { described_class.new(Difficulty.find('easy').level) }
+
+    before do
+      stub_const('Codebreaker::Storage::STATS_DB', './lib/db/test_stats.yml')
+    end
+
+    describe '#hint' do
+      it 'When hints are over' do
+        game.instance_variable_set(:@hints, 0)
+        expect(game.hints).to eq(0)
+      end
+
+      it 'When hints are not over' do
+        game.instance_variable_set(:@hints, 1)
+        expect(game.hints).to eq(1)
+      end
+
+      it 'When hints decrease' do
+        game.instance_variable_set(:@hints, 2)
+        game.hint
+        expect(game.hints).to eq(1)
+      end
+    end
+
+    describe '#guess' do
+      it 'When guess are over' do
+        game.instance_variable_set(:@attempts, 0)
+        expect(game.attempts).to eq(0)
```

Comment:
please test your methods with such approach:
1. set state for instance that you are testing
2. execute methods that you test
3. check that state of your instance was changed or some another method was executed

because such approach doesn't show what really method guess do